tripy/tripy/common/device.py

```diff
@@ -36,7 +36,7 @@ class device:
     kind: str
     index: int
 
-    def __init__(self, device: str) -> None:
+    def __init__(self, device: Union[str, Tuple[str, int]]) -> None:
```

Comment:
Is there a way we could bypass the type checks? Some of the constructors have backdoors like this which are not intended for the end user. It would be good if we could keep the type annotation the same but disable type checking when we use the internal paths. `Tensor` and `DimensionSize` have similar backdoors where the data can be `None`.

Reply:
If it's too difficult to do that, we could maybe look at an alternate approach like using `__new__` in such cases.

Reply:
The question would be how to encode those special cases. We could add an argument to the registry for exceptions, maybe.

Reply:
But we do want type checking to be enabled when a user calls the method. Maybe we can introduce a static method that calls `__new__` and bypasses `__init__` entirely.

Reply:
Went with the static method approach, ends up being very simple.